lib/rubygems/compiler.rb

```diff
@@ -47,6 +47,8 @@ def compile
       gemspec.files.push file
     end
 
+    gemspec.files = gemspec.files.select {|f| File.exists?("#{target_dir}/#{f}") } if @options[:purge]
```

Comment:
I remember it was present:

http://www.rubydoc.info/stdlib/core/1.8.7/Array#reject%21-instance_method

Thank you.